onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,49 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
+/** Check if it will be possible to remove or fuse a node.
     We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+    - The node should not produce a graph output unless something else will produce that output.
```

Comment:
Not sure it is clear what it means that "something else will produce that output."

Reply:
'unless a replacement node or initializer will produce the graph output'?